src/main/java/com/google/sps/servlets/ColumnsFromTablesServlet.java

```diff
@@ -48,35 +47,35 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       String[] listOfTables = request.getParameterValues(constant.TABLE_SELECT_PARAM);
       String database = request.getParameter(constant.DATABASE_PARAM);
 
+      this.dbClient = DatabaseConnector.getInstance().getDbClient(database);
 
-      Spanner spanner = SpannerOptions.newBuilder().build().getService();
-      DatabaseId db = DatabaseId.of("play-user-data-beetle", "test-instance", database);
-      this.dbClient = spanner.getDatabaseClient(db);
+      String query = "";
+      String selectedTables = "";
 
-      if(listOfTables.length == 1){
-          constant.QUERY = constant.QUERY + constant.GET_COLUMNS_FROM_TABLES + "\'" + listOfTables[0] + "\'" + constant.GROUP_BY_TABLE_NAMES;
-      }else{
-          constant.QUERY = constant.QUERY + constant.GET_COLUMNS_FROM_TABLES;
-          for(int i = 0; i < listOfTables.length; i++){
-              if(i != listOfTables.length-1){
-                constant.SELECTED_TABLES = "\'" + listOfTables[i] + "\'";
-                constant.QUERY = constant.QUERY + constant.SELECTED_TABLES + constant.COMMA;
-              }else{
-                  constant.SELECTED_TABLES = "\'" + listOfTables[i] + "\'";
-                  constant.QUERY = constant.QUERY + constant.SELECTED_TABLES;
-                }
-            }
-          constant.QUERY = constant.QUERY + constant.GROUP_BY_TABLE_NAMES;   
+      if (listOfTables.length == 1) {
+        query = query + constant.GET_COLUMNS_FROM_TABLES + "\'" + listOfTables[0] + "\'" + constant.GROUP_BY_TABLE_NAMES;
+      } else {
+        query = query + constant.GET_COLUMNS_FROM_TABLES;
+        for (int i = 0; i < listOfTables.length; i++) {
+          if (i != listOfTables.length-1) {
+            selectedTables = "\'" + listOfTables[i] + "\'";
+            query = query + selectedTables + ", ";
+          } else {
+            selectedTables = "\'" + listOfTables[i] + "\'";
+            query = query + selectedTables;
+          }
         }
```

Comment:
Have you looked at  String.join()? https://docs.oracle.com/javase/8/docs/api/java/lang/String.html. This will be a good use for constructing selectedTables.

Reply:
@hilakey